docs/cudf/source/cudf_polars/api.md

```diff
@@ -0,0 +1,17 @@
+(cudf-polars-api)=
+# API
+
+For the most part, the public API of `cudf-polars` is the polars API. All
+cudf-polars specific configuration is done through options passed to
+{class}`~polars.lazyframe.engine_config.GPUEngine`. The majority of the options
+are passed as `**kwargs` and collected into the configuration described below
+
+
+```{eval-rst}
+.. automodule:: cudf_polars.utils.config
```

Comment:
Feel free to push back on adding any / all of these to the "public" API.

I think it's important to have some coverage of the options since the `StreamingExecutor` is more complicated. Doing through sphinx autodoc would be my preference, but we can described them in prose if folks want.